### PR TITLE
Implement --md5 and --channel-subdir for non-explicit env export

### PR DESCRIPTION
**Pull request by jonashaag**
This keeps the old behaviour of defaulting to md5 output for explicit files and no md5 output for environment files, but allows for outputting md5 with `--md5` for environment files.

### CODE REVIEW

**Comment by JohanMabille** [comment]
I've jsut merged the fix for the `umamba_tests_win_cmd` ci job in the main branch, rebasing your PR should make the CI pass.

**Comment by jonashaag** [comment]
Done, I've also added `--channel-subdir`

**Review by JohanMabille — APPROVED** [review]

File: micromamba/src/env.cpp [hunk 1]
```diff
@@ -123,21 +124,31 @@ set_env_command(CLI::App* com, Configuration& config)
...
+                        if (channel_subdir)
+                        {
+                            dependencies << channel.name() << "/" << v.subdir << "::";
+                        }
+                        dependencies << v.name << "=" << v.version;
                         if (!no_build)
                         {
                             dependencies << "=" << v.build_string;
                         }
+                        if (no_md5 == -1)
```
[the remark this answers is not in the document]
Comment (jonashaag) [reply micromamba/src/env.cpp:145]: I think when you pass `--md5`

[the remark this answers is not in the document]
Comment (AntoinePrv) [reply micromamba/src/env.cpp:145]: I'm also a bit confused why `no_md5` is not a boolean. If there are more complex cases than boolean, I think an enum would be more readable.

[the remark this answers is not in the document]
Comment (jonashaag) [reply micromamba/src/env.cpp:145]: I just experimentally verified the following behavior:

- Passing nothing: `no_md5=0`
- Passing `--no-md5`: `no_md5=1`
- Passing `--md5`: `no_md5=-1`

[the remark this answers is not in the document]
Comment (jonashaag) [reply micromamba/src/env.cpp:145]: I'm merging this for now but I'm happy to improve on it using an enum or something else in a follow up PR

**Pull request merged**